src/ui/topeditorcontainer.cpp

```diff
@@ -89,9 +89,10 @@ void TopEditorContainer::on_currentTabChanged(int index)
     if (!tabWidget)
         return;
 
-    if (m_currentTabWidget == tabWidget && m_currentTabWidget->currentIndex() == index)
+    if (m_currentTabWidget == tabWidget && m_currentTabIndex == index)
```

Comment:
The intend of `m_currentTabWidget->currentIndex() == index` was to check if the old and new tab were the same. But both `currentIndex()` and `index` always contain the new tab index so this part is always true. Therefore, the whole condition will always be true when switching tabs inside the current TabWidget. That led to no `currentTabChanged` signals being emitted at all.

That fixed a bunch of issues with the ui not updating.